cmd/plugin/secret_generation.go

```diff
@@ -0,0 +1,454 @@
+		namespace,
+		"create",
+		"token",
+		serviceAccount,
+		"--duration=8760h",
+	}
+	serviceAccountToken, err := exec.Command(kubectl, args...).Output()
+	if err != nil {
+		if log.V(1).Enabled() {
+			log.V(1).Error(err, "unable to generate service account token from kubeconfig", "context", contextName)
+		}
+		return "", fmt.Errorf("unable to generate service accunt token for %v from kubeconfig", contextName)
+	}
+
+	return string(serviceAccountToken), nil
+}
+
+func defineKubeConfig(log logr.Logger, clusterServer, clusterCA, serviceAccountToken string) Config {
+	log.V(1).Info("Generate a new kubeconfig using the service account token")
+	return Config{
+		APIVersion: "v1",
+		Kind:       "Config",
+		Clusters: []NamedCluster{
+			{
+				Name: name,
+				Cluster: Cluster{
+					Server:                   clusterServer,
+					CertificateAuthorityData: clusterCA,
+				},
+			},
+		},
+		Contexts: []NamedContext{
+			{
+				Name: name,
+				Context: Context{
+					Cluster: name,
+					User:    serviceAccount,
+				},
+			},
+		},
+		CurrentContext: name,
+		Users: []NamedUser{
+			{
+				Name: serviceAccount,
+				User: User{
+					Token: serviceAccountToken,
+				},
+			},
+		},
+	}
+}
+
+func checkKubectl(log logr.Logger) error {
+	log.V(1).Info("Check that kubectl is in the users path")
+	if _, err := exec.LookPath(kubectl); err != nil {
+		if log.V(1).Enabled() {
+			log.V(1).Error(err, "kubectl not found in user path")
+		}
+		return errors.New("kubectl not found in user $PATH")
+	}
+	return nil
+}
+
+func saveKubeConfig(log logr.Logger, kubeConfig Config, dirPath string) (*os.File, error) {
+	log.V(1).Info("Save kubeconfig temporarily for testing")
+	f, err := os.CreateTemp(dirPath, "kubeconfig")
+	if err != nil {
+		log.Error(err, "Unable to create temporary file for sercet")
+		return nil, errors.New("temporary file creation failure.")
+	}
+
+	data, err := yaml.Marshal(kubeConfig)
+	if err != nil {
+		_err := errors.New("Internal error marshaling yaml")
+		err_ := tidyFiles(log, f.Name())
+		if err_ != nil {
+			log.Error(err, "failed to remove temporary kubeconfig")
+			_err = fmt.Errorf(err_.Error(), err)
+		}
+		log.Error(err, "marshaling yaml of kubeConfig file")
+		return nil, _err
+	}
+
+	_, err = f.Write(data)
+	if err != nil {
+		_err := errors.New("Internal error writing file to disc")
+		err_ := tidyFiles(log, f.Name())
+		if err_ != nil {
+			log.Error(err, "failed to remove temporary kubeconfig")
+			_err = fmt.Errorf(err_.Error(), err)
+		}
+		log.Error(err, "error writing of kubeConfig file", "filename", f.Name())
+		return nil, _err
+	}
+
+	return f, nil
+
+}
+
+func verifyKubeConfigConnection(log logr.Logger, kubeConfigPath string) error {
+	log.V(1).Info("Verify the kubeconfig works")
+	args := []string{
+		fmt.Sprintf("--kubeconfig=%v", kubeConfigPath),
+		"version",
+	}
+	cmd := exec.Command(kubectl, args...)
+	_, err := cmd.Output()
+	if err != nil {
+		log.V(0).Info("unable to connect to cluster cluster")
```

Comment:
```suggestion
		log.V(0).Info("unable to connect to cluster")
```

Doesn't add much value since it's the same as the error its going to print, can probably just be removed